tests/chainer_tests/test_gradient_check.py

```diff
@@ -257,6 +259,209 @@ def test_differenct_eps_gpu(self):
+    {'func': 'abs', 'x': [-100.], 'result': False},
+    {'func': 'step', 'x': [0.], 'result': True},
+    {'func': 'step', 'x': [default_eps * 0.8], 'result': True},
+    {'func': 'step', 'x': [-default_eps * 0.8], 'result': True},
+    {'func': 'step', 'x': [default_eps * 1.2], 'result': False},
+    {'func': 'step', 'x': [-default_eps * 1.2], 'result': False},
+    {'func': 'step', 'x': [100.], 'result': False},
+    {'func': 'step', 'x': [-100.], 'result': False},
+    {'func': 'clip', 'x': [0.], 'result': True},
+    {'func': 'clip', 'x': [1.], 'result': True},
+    {'func': 'clip', 'x': [0.5], 'result': False},
+    {'func': 'floor', 'x': [0.], 'result': True},
+    {'func': 'floor', 'x': [100 + default_eps * 0.8], 'result': True},
+    {'func': 'floor', 'x': [100 - default_eps * 0.8], 'result': True},
+    {'func': 'floor', 'x': [100 + default_eps * 1.2], 'result': False},
+    {'func': 'floor', 'x': [100 - default_eps * 1.2], 'result': False},
+    {'func': 'exp', 'x': [-100], 'result': False},
+    {'func': 'exp', 'x': [0.], 'result': False},
+    {'func': 'exp', 'x': [13.], 'result': False},
+    {'func': 'log', 'x': [100.], 'result': False},
+    # (Smaller epsilon is required because slope is steep)
+    {'func': 'log', 'x': [1e-3], 'eps': 1e-6, 'result': False},
+    {'func': 'log', 'x': [0.], 'result': True,
+     'ignore_warning': RuntimeWarning},
+    # (Invalid input domain)
+    {'func': 'log', 'x': [-10.], 'result': False,
+     'ignore_warning': RuntimeWarning},
+    {'func': 'tan', 'x': [default_eps * 1.2], 'result': False},
+    {'func': 'tan', 'x': [default_eps * 0.8], 'result': False},
+    {'func': 'tan', 'x': [math.pi / 2], 'result': True},
+    {'func': 'tan', 'x': [-math.pi / 2], 'result': True},
+    {'func': 'tan', 'x': [3 * math.pi / 2], 'result': True},
+    {'func': 'tan', 'x': [3 * math.pi / 2 + default_eps * 0.8],
+     'result': True},
+    {'func': 'tan', 'x': [3 * math.pi / 2 - default_eps * 0.8],
+     'result': True},
+    # (Smaller epsilon is required because slope is steep)
+    {'func': 'tan', 'x': [3 * math.pi / 2 + 1e-3], 'eps': 1e-6,
+     'result': False},
+    # (Smaller epsilon is required because slope is steep)
+    {'func': 'tan', 'x': [3 * math.pi / 2 - 1e-3], 'eps': 1e-6,
+     'result': False},
+    {'func': 'nan_segment', 'x': [0.], 'result': False},
+    {'func': 'nan_segment', 'x': [-1.], 'result': True},
+    {'func': 'nan_segment', 'x': [1.], 'result': True},
+])
+class NumericalGradientDetectNondifferentiableTest(unittest.TestCase):
+
+    def setUp(self):
+        self.eps = getattr(self, 'eps', default_eps)
+        self.ignore_warning = getattr(self, 'ignore_warning', None)
+
+    def _func_zero(self, x):
+        xp = cuda.get_array_module(x)
+        return xp.zeros_like(x),
+
+    def _func_linear(self, x):
+        return 2 * x,
+
+    def _func_quadratic(self, x):
+        return x * x + 2.,
+
+    def _func_cubic(self, x):
+        return -3 * x ** 3 + 2 * x ** 2 + 1
```

Comment:
Please add a trailing comma to return a tuple, similar to the remaining functions.

Reply:
Thank you. Fixed!